src/hrflow_connectors/connectors/bullhorn/actions.py

```diff
@@ -0,0 +1,227 @@
+                        "date_end") else None
+                }
+                experience_json.append(experience)
+            return experience_json
+
+        def get_attachments(attachment_list):
+            attachments_json = []
+            for hrflow_attachment in attachment_list:
+                url = hrflow_attachment["public_url"]
+                response = requests.get(url)
+                b64 = base64.b64encode(response.content)
+
+                attachment = {
+                    "externalID": "portfolio",
+                    "fileContent": b64.decode(),
+                    "fileType": "SAMPLE",
+                    "name": hrflow_attachment["file_name"],
+                    "contentType": "text/plain",
+                    "description": "Resume file for candidate.",
+                    "type": "cover"
+                }
+                attachments_json.append(attachment)
+            return attachments_json
+
+        enrich_profile_education = get_education(data.get("educations"))
+        enrich_profile_experience = get_experience(data.get("experiences"))
+        enrich_profile_attachment = get_attachments(data.get("attachments"))
+        # When the action needs to send several requests to push a profile
+        # We group the formats of the different requests in a `profile_body_dict`.
+        profile_body_dict = dict(
+            create_profile_body=create_profile_body,
+            enrich_profile_education=enrich_profile_education,
+            enrich_profile_experience=enrich_profile_experience,
+            enrich_profile_attachment=enrich_profile_attachment,
+        )
+        return profile_body_dict
+
+    def push(self, data):
+        profile_body_dict = next(data)
+        create_profile_body = profile_body_dict["create_profile_body"]
+        enrich_profile_education = profile_body_dict["enrich_profile_education"]
+        enrich_profile_experience = profile_body_dict["enrich_profile_experience"]
+        enrich_profile_attachment = profile_body_dict["enrich_profile_attachment"]
+
+        # Preparing the request to push the profile
+        session = requests.Session()
+        push_profile_request = requests.Request()
+        push_profile_request.method = "PUT"
+        push_profile_request.url = f"https://{self.subdomain}.bullhornstaffing.com/rest-services/7zwdd0/entity/Candidate"
+        push_profile_request.auth = self.auth
+        push_profile_request.json = create_profile_body
+        push_profile_request.headers = {"content-type": "application/json"}
+        prepared_request = push_profile_request.prepare()
+
+        # Send request
+        response = session.send(prepared_request)
+        if not response.ok:
+            error_message = "Unable to push the data ! Reason : `{}`,`{}`"
+            raise RuntimeError(error_message.format(response.status_code,response.content))
```

Comment:
Instead of RuntimeError use the nex hrflow custom errors created by @CorentinDuchene  
https://github.com/Riminder/hrflow-connectors/blob/dev/src/hrflow_connectors/core/error.py

Reply:
Pour cela, il faut qu'il merge `dev` -> sa branche (et surtout pas l'inverse) pour avoir les dernières mises à jour et adapter sa branche à toutes les modifications faites depuis (le renommage de certaines classes).